arch/arm/src/arm/arm_syscall.c

```diff
@@ -118,11 +122,6 @@ uint32_t *arm_syscall(uint32_t *regs)
        */
 
       case SYS_switch_context:
-        {
-          DEBUGASSERT(regs[REG_R1] != 0 && regs[REG_R2] != 0);
```

Comment:
Is this case supposed to be empty ?

Reply:
In some arch, empty implementations are acceptable.
The line *(uint32_t **)regs[REG_R1] = regs; is intended to save the context, which can also be achieved through (*running_task)->xcp.regs = regs;
The line up_set_current_regs((uint32_t *)regs[REG_R2]); sets the context to be returned to, which can be directly using tcb->xcp.regs.